[Hub Generated] Review request for Microsoft.Consumption to add version 2019-01-01

shbha opened the pull request:
If you are a MSFT employee you can view your [work branch via this link](https://portal.azure-devex-tools.com/app/branch/shbha/azure-rest-api-specs/dev-consumption-Microsoft.Consumption-2019-01-01).

### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

shbhamsft committed:
Adds base for updating Microsoft.Consumption from version stable/2018-10-01 to version 2019-01-01

shbhamsft committed:
Updates readme

shbhamsft committed:
Updates API version in new specs and examples

shbhamsft committed:
updated readme to support new stable version 2019-01-01

AutorestCI commented:
# Automation for azure-sdk-for-js

AutorestCI commented:
# Automation for azure-sdk-for-python

AutorestCI commented:
# Automation for azure-sdk-for-ruby

AutorestCI commented:
# Automation for azure-sdk-for-go

shbha commented:
+ @veronicagg

AutorestCI commented:
# Automation for azure-sdk-for-java

azuresdkci commented:
Can one of the admins verify this patch?

AutorestCI commented:
# Automation for azure-sdk-for-node

veronicagg commented:
Taking this PR as I was reviewing the original, which is getting abandoned.

ravbhatnagar commented:
Doesnt look like any changes other than api version change. Signing off from ARM

veronicagg commented on specification/consumption/resource-manager/readme.md:
````diff
@@ -335,4 +354,3 @@ java:
 regenerate-manager: true
 generate-interface: true
 ```
-
````
@shbha could you add/update readme file for Java code generation in this PR too? Thanks!

veronicagg reviewed (changes requested):
Just one comment to update Java code generation in reamde.md file, rest looks good! Thanks for submitting a new PR with clearer commits.

shbhamsft committed:
version introduced for java readme.md

veronicagg committed:
Updating swaggeer to SDK section for java SDK generation

veronicagg commented:
@anuchandy could you help taking a look at the failure for Java generation? Java was missing from swagger to SDK configuration and as I added it, I'm seeing it failing. Thanks!

veronicagg commented:
@shbha I noticed Java SDK wasn't getting generated, I added the language to the readme.md configuration and it's now producing a failure, so would like to hear back from one of the Java SDK owners. After solving that failure, we should be good to merge this PR.

shbha commented:
@veronicagg - Delay merge or PR. Upon internal review of this PR there is a requirement to introduce additional changes. 1) define and use method.parameter scope and introduce a few new scopes for existing methods. Arm provided this review on a different PR ( for a different specification RP) and I would like to follow best practices here as well. I would really appreciate it if we could hold off on merge till these changes are made.

veronicagg commented:
@shbha sure, thanks for letting me know, I've edited the title to indicate this. In the meantime, we're investigating the Java SDK failure generation. Please keep me posted on progress, thanks!

shbha1 committed:
introducing MG scope for tags/budgets
introduce scope (method) parameter for APIs that support numerous scopes (arm-team suggestion implemented)

shbha commented:
The required changes have been made @veronicagg. Thank you

veronicagg commented on specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-01-01/consumption.json:
```diff
@@ -1,4547 +1,2930 @@
-            "description": "Error response describing why the operation failed.",
-            "schema": {
-              "$ref": "#/definitions/ErrorResponse"
+    "swagger": "2.0",
```
Looks like the content got indented 4 spaces instead of 2, so the diff is really hard to see in this comment, as most looks new, could you update indentation?

shbha replied:
hi @veronicagg , could we change diffsettings to ignore whitespace? I tried it, and I was able to view the specific changes. The indentation is 4 spaces in our other RP (cost.management) and this will introduce consistency. Sorry for the inconvenience.

shbha replied:
@veronicagg - apparently the space was set to 2 for the other RP as well. I've updated the indentation back to 2.

anuchandy commented:
@shbha could you please merge this PR https://github.com/shbha/azure-rest-api-specs/pull/1? This fixes Java code generation for previous api-version.

anuchandy committed:
Fixing Java code-generation (#1)

* Fixing Java code-generation

* Removing duplicate directive from java section

shbha1 committed:
update indentation

veronicagg reviewed (approved):
Thanks for the updates.
